client/lib/signup/flow-controller.js

```diff
@@ -100,6 +100,8 @@ assign( SignupFlowController.prototype, {
 
 	_resetStoresIfUserHasLoggedIn: function() {
 		if (
+			// TODO: Address this properly if we move forward with the user step being last
+			this._flowName !== 'onboarding-user-last' &&
```

Comment:
The A/B test to move the user step has wrapped up (https://github.com/Automattic/wp-calypso/pull/35552), so this line can be removed now.